tests/serialization/test_dag_serialization.py

```diff
@@ -135,7 +136,7 @@
                 'label': 'custom_task',
             },
         ],
-        "timezone": "UTC",
+        "schedule_timezone": "UTC",
```

Comment:
Okay, so now we come to an interesting problem that we, so far, have not had to deal with.

We need to be able to continue to parse "old" serialized versions, so this might be easiest to _not_ change the field name, and in the serialized representation keep it called `timezone`, but just upgrade the type to allow it to become int or string.

Reply:
Ok, I managed to keep as `timezone` in the serialized object, but this getting more and more complex , and more hacky that the alternative #16631.

Reply:
If we can find a to not have to do `return pendulum.parse("2021-01-01T12:00:00" + tz_name).tzinfo` I'd be happy with that approach, but with that it just feels too hacky I think.